client/my-sites/subscribers/mutations/use-subscriber-remove-mutation.ts

```diff
@@ -107,16 +108,22 @@ const useSubscriberRemoveMutation = (
 						await wpcom.req.post(
 							`/sites/${ siteId }/email-followers/${ emailSubscriptionId }/delete`
 						);
-						wasRemoved = true;
+						emailRemoved = true;
 					} catch ( e ) {
 						// Only throw if we haven't successfully removed them through any other method.
-						if ( ! wasRemoved ) {
+						if ( ! wpcomRemoved ) {
 							throw new Error( ( e as ApiResponseError )?.message );
 						}
 					}
 				}
 
-				return wasRemoved;
+				// Consider removal successful if:
+				// 1. We removed the email subscription (if it existed)
+				// 2. We removed the wpcom following (if it existed)
+				const emailSuccess = ! emailSubscriptionId || emailRemoved;
+				const wpcomSuccess = ! isNaN( numericUserId ) ? wpcomRemoved : true;
```

Comment:
I think we can simplify it a bit.

```suggestion
				const emailSuccess = emailSubscriptionId ? emailRemoved : false;
				const wpcomSuccess = isNaN( numericUserId ) ? true : wpcomRemoved;
```

Reply:
@mehmoodak Thanks for reviewing! I think there's a slight difference in the logic. Current version:

"Success if there was no subscription ID OR if we successfully removed it"

Suggestion:

"Success only if there was a subscription ID AND we removed it"

Reply:
My mistake .... we can switch the condition to

```
const emailSuccess = emailSubscriptionId ? emailRemoved : true;
```

The change is good either way but I think its easier to read the code if both are ternary expressions 🙂

Reply:
Got it! Updated in https://github.com/Automattic/wp-calypso/pull/103306/commits/be7978f5837baa25e5338b8336e1c6fd98b4b31a